src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -2721,6 +2727,70 @@ export class ResultsViewPageStore {
         },
     });
 
+    readonly fusions = remoteData<StructuralVariant[]>({
+        await: () => [
+            this.genes,
+            this.selectedMolecularProfiles,
+            this.samples
+        ],
+        invoke: () => {
+            return client.fetchStructuralVariantsUsingPOST({
+                'structuralVariantFilter': getParamsForStructuralVariants(this.genes.result, this.selectedMolecularProfiles.result, this.samples.result)
+            })
+        }
+    });
+
+    readonly fusionsByGene = remoteData<{[hugeGeneSymbol: string]: StructuralVariantExt[]}>({
+        await: () => [this.genes, this.fusions],
+        invoke: () => {
+            const fusionsByGene = _(this.genes.result!)
```

Comment:
might be better to move longer logic into some utility function

Reply:
good point! Will change.